pkg/ccl/backupccl/backuputils/utils.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Licensed as a CockroachDB Enterprise file under the Cockroach Community
+// License (the "License"); you may not use this file except in compliance with
+// the License. You may obtain a copy of the License at
+//
+//     https://github.com/cockroachdb/cockroach/blob/master/licenses/CCL.txt
+
+package backuputils
+
+import (
+	"net/url"
+	"path"
+
+	"github.com/cockroachdb/cockroach/pkg/cloud"
+)
+
+// URLSeparator represents the standard separator used in backup URLs.
+const URLSeparator = '/'
+
+// RedactURIForErrorMessage redacts any storage secrets before returning a URI which is safe to
+// return to the client in an error message.
+func RedactURIForErrorMessage(uri string) string {
+	redactedURI, err := cloud.SanitizeExternalStorageURI(uri, []string{})
+	if err != nil {
+		return "<uri_failed_to_redact>"
+	}
+	return redactedURI
+}
+
+// JoinURLPath forces a relative path join by removing any leading slash, then
+// re-prepending it later.
+//
+// Stores are an odd combination of absolute and relative path.
+// They present as absolute paths, since they contain a hostname. URL.Parse
+// thus prepends each URL.Path with a leading slash.
+// But some schemes, e.g. nodelocal, can legally travel _above_ the ostensible
+// root (e.g. nodelocal://0/.../). This is not typically possible in file
+// paths, and the standard path package doesn't like it. Specifically, it will
+// clean up something like nodelocal://0/../ to nodelocal://0. This is normally
+// correct behavior, but is wrong here.
+//
+// In point of fact we block this URLs resolved this way elsewhere. But we
+// still want to make sure to resolve the paths correctly here. We don't want
+// to accidentally correct an unauthorized file path to an authorized one, then
+// write a backup to an unexpected place or print the wrong error message on
+// a restore.
+func JoinURLPath(args ...string) string {
```

Comment:
I could see this living in the cloud package since it is particular to our external storage implementation.  But, we can leave that bikeshed to a follow-up PR if anyone else agrees.